frontend/ucc_gui/src/components/Project/ProjectChallenge3/LearnNewSkillComponent.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import {Player} from "video-react";
+import Upload_video from "../../../image/Settings_Camera.png";
+import "../../../containers/Projects/LearnNewSkill/LearnNewSkill.css";
+import Button from "react-bootstrap/Button";
+import axiosConfig from "../../../axiosConfig";
+import * as cookie from "react-cookies";
+
+
+class LearnNewSkillComponent extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            NewSkill: '',
+            Description: '',
+            Video: '',
+            finalVideo: '',
+            ProjectId: this.props.id,
+            Email: cookie.load('user_emailid')
+        }
+    };
+
+    defaultIfEmpty(value){
+        return value === "" ? "":value;
+    }
+
+    changeHandler = (event) =>{
+        this.setState({
+            ...this.state,
+            [event.target.name]: event.target.value
+        })
+    };
+
+    videoHandler = (event) =>{
+       this.setState({
+           Video : URL.createObjectURL(event.target.files[0]),
+           finalVideo: event.target.files[0]
+        });
+    };
+
+
+    saveHandler(event,  requestType, id){
+        event.preventDefault();
+        let form_data = new FormData();
+        try {
+            form_data.append('NewSkill', this.state.NewSkill);
+            form_data.append('Description', this.state.Description);
+            if (this.state.finalVideo) {
+                form_data.append('Video', this.state.finalVideo, this.state.finalVideo.name);
+            }
+            form_data.append('ProjectId', this.state.ProjectId);
+            form_data.append('Email', this.state.UserEmailId);
+        } catch(err) {
+            console.log(err)
+        }
+
+        switch( requestType ) {
+            case 'post':
+            return axiosConfig.post('http://127.0.0.1:8000/learn_new_skill/', form_data,
```

Comment:
1. No to give the base URL (http://127.0.0.1:8000/), just give learn_new_skill/, 
2. also add 'charityproject' because that's the complete URL of the back-end API
```suggestion
            return axiosConfig.post('charityproject/learn_new_skill/', form_data,
```

Reply:
1. Made the changes.
2. Will implement it at the backend.